### ISSUE TITLE
Process Babel `main` option

### ISSUE THREAD

**Issue by fh-igd-mueller-roemer** [valentjn/vscode-ltex#391]
**Is your feature request related to a problem? Please describe.**
While parsing Babel package options, LTeX ignores the `main` option and instead only uses the order of the language options to select the main document language. For example, a document with

```latex
\usepackage[USenglish, main=ngerman]{babel} % or equivalently \usepackage[main=ngerman, USenglish]{babel}
```

is checked as `en-US`, despite the main language for Babel being set to `de-DE` (`ngerman`).

**Describe the solution you'd like**
LTeX parses and recognizes the `main` option and uses it - instead of language order - to select the main language if present.

**Describe alternatives you've considered**

Option A:
The other option to specify a main language in Babel is to pass the main language *last*:

```latex
\usepackage{USenglish, ngerman}{babel}
```

This method is parsed correctly by LTeX, however, this way of specifying the main language is less preferable and less beginner friendly, as it requires implicit knowledge of the meaning of the order of options in Babel. Furthermore, the use of a tool like LTeX should ideally not require writing ones LaTeX code differently.

Option B:
Another option may be to prioritize the `% LTeX: language=` magic comment over the document language set by parsing Babel language options.

**Comment by valentjn** [comment]
Implemented. Note that using `main` to set the main language explicitly is discouraged by the babel authors themselves, except if actually necessary. And of course you need a basic knowledge of how to use a package in order to use it. See [babel manual](http://mirrors.ctan.org/macros/latex/required/babel/base/babel.pdf), Sec. 1.2:

> You can also set the main language explicitly, but it is discouraged except if there \[sic] a real reason to do so:
>
> ```latex
> \documentclass{article}
> \usepackage[main=english,dutch]{babel}
> ```
>
> Examples of cases where `main` is useful are the following.
>
> NOTE Some classes load babel with a hardcoded language option. Sometimes, the main language can be overridden with something like that before `\documentclass`:
>
> ```latex
> \PassOptionsToPackage{main=english}{babel}
> ```
>
> WARNING Languages may be set as global and as package option at the same time, but in such a case you should set explicitly the main language with the package option `main`:
>
> ```latex
> \documentclass[italian]{book}
> \usepackage[ngerman,main=italian]{babel}
> ```

**Comment by fh-igd-mueller-roemer** [comment]
I personally don't agree with then discouragement, but in any case, the last example is actually what I have right now, so my actual workaround was closer to

```
\documentclass[ngerman]{book}
\usepackage[USenglish, ngerman, main=ngerman]{babel}
```

Thank you very much!

**Comment by valentjn** [comment]
Feature released in 12.0.0.
